src/Verifier/SAW/Simulator/What4/SWord.hs

```diff
@@ -360,6 +364,21 @@ bvCountLeadingZeros = bvUn W.bvCountLeadingZeros
 bvCountTrailingZeros :: SWordUn
 bvCountTrailingZeros = bvUn W.bvCountTrailingZeros
 
+bvForall :: forall sym. W.IsSymExprBuilder sym =>
+  sym -> Natural -> (SWord sym -> IO (Pred sym)) -> IO (Pred sym)
+bvForall sym w f =
+  case W.userSymbol "i" of
+    Left err -> fail $ show err
+    Right indexSymbol ->
+      case someNatVal w of
+        SomeNat (Proxy :: Proxy w) -> if
+          | 1 <= w
+          , Refl :: (1 <=? w) :~: 'True <- unsafeCoerce Refl -> do
```

Comment:
```
case mkNatRepr n of
  Some w
    | Just LeqProof <- testLeq (knownNat @1 ) w -> ...
    | otherwise -> ...
```
See https://hackage.haskell.org/package/parameterized-utils-2.0/docs/Data-Parameterized-NatRepr.html#v:testLeq

Reply:
Neat, this is exactly what I was looking for. Thanks!